src/document/range.js

```diff
@@ -72,6 +72,211 @@ CKEDITOR.define( [ 'document/positioniterator', 'document/position' ], ( Positio
 		}
 
 		/**
+		 * Checks whether this {document.Range range} intersects with the given {document.Range range}.
+		 *
+		 * @param {document.Range} otherRange Range to check.
+		 * @returns {boolean} True if ranges intersect.
+		 */
+		intersectsWith( otherRange ) {
```

Comment:
We do not need `intersectsWith` when you can check exactly the same using `getCommon`.